idx/cassandra/cassandra.go

```diff
@@ -264,10 +264,36 @@ func (c *CasIdx) Stop() {
 	c.session.Close()
 }
 
-func (c *CasIdx) AddOrUpdate(data *schema.MetricData, partition int32) idx.Archive {
+func (c *CasIdx) UpdateMaybe(point schema.MetricPoint, partition int32) (idx.Archive, bool) {
 	pre := time.Now()
-	existing, inMemory := c.MemoryIdx.Get(data.Id)
-	archive := c.MemoryIdx.AddOrUpdate(data, partition)
+
+	// note that both functions return an 'ok' bool.
+	// abeit very unlikely,
+	// the idx entry could be pruned in between the two calls and so they could be different
+	existing, inMemory := c.MemoryIdx.Get(point.MKey)
+	archive, inMemory2 := c.MemoryIdx.UpdateMaybe(point, partition)
+
+	if !updateCassIdx {
+		statUpdateDuration.Value(time.Since(pre))
+		return archive, inMemory2
+	}
+
+	if inMemory {
+		c.deleteOldIfMoved(existing, partition)
+	}
+
+	if inMemory2 {
+		archive = c.updateCassandraIfStale(inMemory, archive, partition)
```

Comment:
> One example we have is that we don't include the Interval in the id we generate (for reasons I can explain if you are curious).

always :)

I think I don't really understand the question/problem,
but there's 2 places where this logic is called (UpdateMaybe which takes the MetricPoint, and AddOrUpdate which takes a full MetricData). however, by the time we get to the updateCassandraIfStale call, it doesn't really matter which of the 2 paths we took to get to this logic, as the calls use the same parameters which it pulled out of the memory index (possibly after an update), and the partition value.
In that regard, whether you sent a MetricData (with the same interval) or a MetricPoint, the logic should be equivalent.
and the ordering should remain honored throughout the kafka->CassandraIndex->Cassandra pipeline. 

does this help @shanson7 ? otherwise let's discuss further

Reply:
My original comment can be ignored. As long as any data that is not encoded into the `Id` is in the MetricPoint, then this should be sufficient.

Reply:
there is one field from MetricData that doesn't contribute to its Id, which is `Name`. However, since we always say that field should equal the `Metric` field (which does contribute the id), this should be OK. Note that because of this, we plan to remove one of these two as they should always be the same.

Reply:
Yes, I noticed that. I use `Name` when encoding and set `Metric` to `.` for everything (to reduce the size of the messages)